test/extended/etcd/vertical_scaling.go

```diff
@@ -186,4 +201,117 @@ var _ = g.Describe("[sig-etcd][Feature:EtcdVerticalScaling][Suite:openshift/etcd
+	//
+	// 1) If the CPMS is active, first disable it by deleting the CPMS custom resource
+	// 2) Delete a machine
+	// 3) Ensure the ready master node count and the voting member count remain at 3 after the
+	// 	  deletion of a machine and before a new machine is added, to verify that scale-down hasn't occurred.
+	// 4) Create a new master machine that belongs to the same index as the deleted machine
+	// 5) Scale-up happens first before the scale-down
+	// 6) Validate scales-up happens first by verifying 4 voting members in etcd cluster
+	// 7) Then scale-down is validated by confirming the member removal and changes in the cluster membership
+	g.It("is able to vertically scale up and down when CPMS is disabled [apigroup:machine.openshift.io]", func() {
+		if cpmsActive {
+			// step 0: disable the CPMS
+			framework.Logf("Disable the CPMS")
+			err := scalingtestinglibrary.DisableCPMS(ctx, g.GinkgoT(), cpmsClient)
+			err = errors.Wrap(err, "pre-test: failed to disable the CPMS")
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			// re-enable CPMS after the test
+			defer func() {
+				framework.Logf("Re-enable the CPMS")
+				err := scalingtestinglibrary.EnableCPMS(ctx, g.GinkgoT(), cpmsClient)
+				err = errors.Wrap(err, "post-test: failed to re-enable the CPMS")
+				o.Expect(err).ToNot(o.HaveOccurred())
+			}()
+		}
+
+		framework.Logf("CPMS is disabled. The test will delete an existing machine and manually create a new machine to validate scale-down doesn't happen before scale-up event")
+
+		// step 1: delete a running machine
+		deletedMachineName, err := scalingtestinglibrary.DeleteSingleMachine(ctx, g.GinkgoT(), machineClient)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		framework.Logf("Deleted machine %q", deletedMachineName)
+
+		// step 2: Verify the ready master node count and the voting member count remain at 3 after the
+		// deletion of a machine and before a new machine is added to ensure scale-down hasn't occurred.
+		//
+		// Poll for 30 seconds to ensure that the ready master node count hasn't changed from 3. This
+		// will also allow some time before verifying the voting member count remains the same.
+		// After a machine is marked for deletion, the node will remain ready until the deletion hook is removed and
+		// it is drained. The deletion hook will not be removed from the deleting node until a new member joins the cluster.
+		framework.Logf("Ensuring the ready master node count remains at 3 to confirm that scale-down hasn't occurred")
+		o.Expect(wait.PollUntilContextTimeout(ctx, 10*time.Second, 30*time.Second, false, func(ctx context.Context) (done bool, err error) {
+			if err = scalingtestinglibrary.EnsureReadyMasterNodes(context.Background(), 3, nodeClient); err != nil {
+				return true, nil // stop polling if the ready master node count changes
+			}
+			return false, nil
+		})).To(o.HaveOccurred(), "ready master node count shouldn't change from 3, as scale-down isn't expected to happen")
```

Comment:
Functionally this may seem like a good proxy for seeing that we haven't scaled down, but looking closer at `EnsureReadyMasterNodes ` that's actually only checking whether the Nodes are still reporting as ready.
https://github.com/openshift/origin/blob/efdd1fec5bcb3d57687f12117739778574ed8336/test/extended/etcd/helpers/helpers.go#L344C32-L344C41
https://github.com/openshift/origin/blob/f25349c0ca7cd77807b5299772f19ce2631adb43/vendor/k8s.io/api/core/v1/types.go#L6013-L6014

The node may still be ready even if the etcd pod for that node is crashlooping because it's not in the membership.

Instead what we're more concerned with at this point is that the clustermemberremoval controller hasn't actually removed an etcd member to change the membership from 3 to 2.
https://github.com/openshift/cluster-etcd-operator/blob/fc54f7fd4773418e040eef97a3ae13cdffc3872c/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go#L232-L238

So you might want to do something similar to this to check
https://github.com/openshift/origin/blob/efdd1fec5bcb3d57687f12117739778574ed8336/test/extended/etcd/helpers/helpers.go#L374-L399

So you might want to create a new helper to poll that the etcd membership stays at the expected count for 30s or so.

Reply:
I have created a new helper function that polls for 2 minutes to ensure that the etcd membership stays at the expected count. I set the polling time as 2 minutes because, while checking another case it took sometime between 1-1.5 minutes for the member to be removed from the cluster[Remove a pod manifest, then delete the machine hosting the node from which the pod was removed].  If you suggest 2 minutes is too long to wait, I can change it.